File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    length = float('inf')
    n = len(sequence)
    left = 0
    right = 0
    _sum = 0
    answer = []
    while right < n:
        _sum += sequence[right]
        while _sum >= k:
            if _sum == k and length > right - left:
                length = right - left
                answer = [left, right]
            _sum -= sequence[left]
            left += 1
        right += 1
    
    return answer

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
import java.util.*;

class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = {};
        List<int[]> idx = new ArrayList<>();
        int left = 0;
        int right = 0;
        int sum = sequence[0];
        while (right < sequence.length) {
            if (sum == k) {
                idx.add(new int[]{left, right});
                sum -= sequence[left++];
            }
            else if (sum < k) {
                right++;
                if (right < sequence.length)
                    sum += sequence[right];
            }
            else {
                sum -= sequence[left++];
            }
        }
        
        int max = Integer.MAX_VALUE;
        for (int[] i : idx) {
            int l = i[0];
            int r = i[1];
            int diff = r - l;
            if (diff < max) {
                answer = i;
                max = diff;
            }
        }
        
        return answer;
    }
}